src/playlist-selectors.js

```diff
@@ -103,91 +125,118 @@ export const comparePlaylistResolution = function(left, right) {
 
   rightWidth = rightWidth || window.Number.MAX_VALUE;
 
+  const leftBandwidth = getBandwidth(left, false);
+  const rightBandwidth = getBandwidth(right, false);
+
   // NOTE - Fallback to bandwidth sort as appropriate in cases where multiple renditions
-  // have the same media dimensions/ resolution
-  if (leftWidth === rightWidth &&
-      left.attributes.BANDWIDTH &&
-      right.attributes.BANDWIDTH) {
-    return left.attributes.BANDWIDTH - right.attributes.BANDWIDTH;
+  // have the same media dimensions/ resolution.
+  if (leftWidth === rightWidth && leftBandwidth && rightBandwidth) {
+    return leftBandwidth - rightBandwidth;
   }
   return leftWidth - rightWidth;
 };
 
-const simpleSelector = function(master, bandwidth, width, height) {
+const simpleSelector = function(master,
+                                bandwidth,
+                                width,
+                                height,
+                                useAverageBandwidth = false) {
   let sortedPlaylists = master.playlists.slice();
-  let bandwidthPlaylists = [];
-  let bandwidthBestVariant;
-  let resolutionPlusOne;
-  let resolutionBestVariant;
-  let haveResolution;
-  let resolutionPlusOneList = [];
-  let resolutionPlusOneSmallest = [];
-  let resolutionBestVariantList = [];
 
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+  // convert the playlists to an intermediary representation to make comparisons easier
+  // and prevent us from re-determining bandwidth each time
+  let sortedPlaylistReps = sortedPlaylists.map((playlist) => {
```

Comment:
since we are no longer sorting the `sortedPlaylists` itself, this can just be `master.playlists.map(...` without affecting the sorting of the original list`